core/src/parking_lot.rs

```diff
@@ -638,6 +685,12 @@ where
 
         // Now we need to check again if we were unparked or timed out. Unlike the
         // last check this is precise because we hold the bucket lock.
+        //
+        // REVIEW: this branch isn't quite clear to me, why is this needed in
+        // addition with `unparked` above? This is only used in the case that
+        // `park_until` returns `false`, but doesn't that mean (given the
+        // platform-specific implementations) that the timeout elapsed so
+        // `timed_out` should be asserted to be true?
```

Comment:
While the timeout may have elapsed, our thread is still in the hash table so another may have unparked it after we timed out but before we called `lock_bucket_checked`. In this case we need to report an `Unparked` with the token that was given to us by the other thread.

Reply:
Ah right yeah, can that be added to the comment?